hadoop-hdfs-project/hadoop-hdfs-rbf/src/test/java/org/apache/hadoop/hdfs/server/federation/store/driver/TestStateStoreDriverBase.java

```diff
@@ -234,6 +234,25 @@ public <T extends BaseRecord> void testInsert(
     assertEquals(11, records2.size());
   }
 
+  public <T extends BaseRecord> void testInsertWithErrorDuringWrite(
```

Comment:
To test my[ above comment](https://github.com/apache/hadoop/pull/5145/files#r1028382408), you would need two iterations of the loop where the first is a failure and the second is a success. The test passes despite the issue because there is only a single write.

Reply:
Yeah, this test doesn't capture the error I'd made. I tried changing the test to do a failure and then a success but I can't get mockito to work right. 
Since I'm working with a spy I need to use the doThrow().when(...) syntax. However, to have a change of behavior I believe mock wants me to use when(...).doThrow().doCallRealMethod().

@mkuchenbecker If it is okay, I'll leave the test as it is. It exposes the original bug in the code and validate that my patch address it.